### PR TITLE
Alex krasn/hotfix mime undefined issue

### PR DESCRIPTION
**Pull request by alex-krasn**

### CODE REVIEW

File: src/services/assetService.ts [hunk 1]
```diff
@@ -192,8 +192,8 @@ export class AssetService {
             if (!types) {
                 console.error(interpolate(strings.editorPage.assetWarning.incorrectFileExtension.failedToFetch, { fileName: corruptFileName.toLocaleUpperCase() }));
             }
-            // If file was renamed/spoofed - fix file extension to true MIME type and show message
-            else if (!types.includes(assetFormat)) {
+            // If file was renamed/spoofed - fix file extension to true MIME if it's type is in supported file types and show message
+            else if (types[0] && !types.includes(assetFormat)) {
```
Comment (XinxingLiu) [inline src/services/assetService.ts:196]: Is it safer to check the length of array?

File: src/services/assetService.ts [hunk 2]
```diff
@@ -192,8 +192,8 @@ export class AssetService {
...
                 assetFormat = types[0];
                 console.error(`${strings.editorPage.assetWarning.incorrectFileExtension.attention} ${corruptFileName.toLocaleUpperCase()} ${strings.editorPage.assetWarning.incorrectFileExtension.text} ${corruptFileName.toLocaleUpperCase()}`);
```
Comment (XinxingLiu) [inline src/services/assetService.ts:198]: If my understanding is correct, the error should be logged whenever types.includes(assetFormat) == false, while assetFormat overwritten should only happen if types is not empty.

**Review by XinxingLiu — APPROVED** [review]

File: src/services/assetService.ts (hunk 2 above)
Comment (alex-krasn) [reply src/services/assetService.ts:198]: yes

File: src/services/assetService.ts [hunk 3]
```diff
@@ -182,7 +182,7 @@ export class AssetService {
             if (nodejsMode) {
                 const FileType = require('file-type');
                 const fileType = await FileType.fromFile(normalizedPath);
-                types = [fileType.ext];
+                types = fileType?.ext;
```
Comment (stew-ro) [inline src/services/assetService.ts:185]: variable "types" should still be an array. 
I think `types = fileType?.ext ? [fileType.ext] : []` is what you want

**Review by stew-ro — COMMENTED** [review]
This PR has issues running ocr on local files with undefined mime types. I have created a PR which resolves this and also uses the `file-type` library. https://github.com/microsoft/OCR-Form-Tools/pull/636/files

I think it may prevent future mime types issues and we should review instead

File: src/services/assetService.ts (hunk 1 above)
Comment (alex-krasn) [reply src/services/assetService.ts]: done

**Pull request closed without merging**